packages/cozy-client/src/hooks/useSafeState.jsx

```diff
@@ -0,0 +1,20 @@
+import { useState, useCallback } from 'react'
+
+import useIsMounted from './useIsMounted'
+
+const useSafeState = initialState => {
+  const mounted = useIsMounted()
+  const [state, setState] = useState(initialState)
+
+  const setSafeState = useCallback(
+    newState => {
+      if (mounted.current) {
+        setState(newState)
+      }
+    },
+    [] // eslint-disable-line react-hooks/exhaustive-deps
+  )
+  return [state, setSafeState]
```

Comment:
nit: testing this hook would document why it is suppopsed to do.

(and why we couldn't use a useState hook)

Reply:
je ne sais pas pourqoi un useState ne suffit pas, c'est du code rapatrié tel que de Banks

Reply:
nit: est-ce qu'on pourrait vérifier si ça marche avec uniquement un useState 🙏 ?

(je pense que le meilleur moment pour investiguer ceci, c'est maintenant, mais si tu n'as pas le temps, tu peux créer une issue ?)

Reply:
Si j'ai le temps là, je vais regarder

Reply:
Initialement on a fait un hook simple : https://github.com/cozy/cozy-banks/commit/9b30b7bc52894f2a0503b74e98794d6dddfe0995 (`useFullyLoadedQuery.jsx`)

Puis on a ajouté la notion de `isFetching` pour éviter de redéclencher le hook alors qu'il est en train de travailler : https://github.com/cozy/cozy-banks/commit/d5984880387486246c7417202dc8a7524e9371de . On ajoute également ici la notion de `isMounted` nécessaire visiblement pour ne setter le isFetching à false que si le hook est bien monté.

Enfin on fusionne ici https://github.com/cozy/cozy-banks/pull/2160/commits/201df73923f065d350a0e8ecc2ca2f3062a0c9d6 l'approche `isMounted` avec le `useState`

J'ai remplacé le `useSafeState` par un `useState`. Dans CCO2 ça fonctionne pareil visiblement, mais dans Banks ça plante... Pour avoir essayé de faire un hook similaire vendredi dernier (ce qui est bien galère) et le use case Banks, je suis plutôt partisan de laisser en l'état étant donné que l'ajout de code n'est pas énorme d'une part, et qu'il n'ajoute qu'un garde-fou supplémentaire finalement.

Reply:
merci